### PR TITLE
Add new OTel pipeline for scraping Kueue metrics and publishing to CloudWatch

### PR DESCRIPTION
**Pull request by rvasahu-amazon**
# Description of the issue

As title says. This PR implements a new translator pipeline for collecting Kueue metrics and exporting them to CloudWatch.
The pipeline uses:
- A custom receiver intended to run alongside the AWS Container Insights Receiver. This new receiver defines a Prometheus scrape config and passes it to a lower-level Prometheus Receiver.
- A new processor.
- The AWS EMF exporter, for which a new configuration has been added.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Added unit testing for new components, updated unit testing for existing ones.
Integration tests in a [separate PR](https://github.com/aws/amazon-cloudwatch-agent-test/pull/428).

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -38,6 +38,7 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/processor/reso
 
 replace (
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver => github.com/amazon-contributing/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver v0.0.0-20241104203805-20919412150d
+	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightskueuereceiver => github.com/amazon-contributing/opentelemetry-collector-contrib/receiver/awscontainerinsightskueuereceiver v0.0.0-20241112181539-bc8278558352
```
Comment (chadpatel) [inline go.mod:41]: let's get this in aws-cwa-dev

Comment (rvasahu-amazon) [reply go.mod:41]: Done, see [this PR](https://github.com/amazon-contributing/opentelemetry-collector-contrib/pull/253).

File: go.mod [hunk 2]
```diff
@@ -208,6 +209,8 @@ require (
 	k8s.io/klog/v2 v2.120.1
 )
 
+require github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightskueuereceiver v0.0.0-20241112181539-bc8278558352
```
Comment (sky333999) [inline go.mod:212]: Can you move this up along with the rest?

File: go.mod (hunk 1 above)
Comment (sky333999) [reply go.mod:41]: You need to update these versions for all components post the merge to aws-cwa-dev.

File: go.mod (hunk 2 above)
Comment (rvasahu-amazon) [reply go.mod]: Done.

File: go.mod (hunk 1 above)
Comment (sky333999) [reply go.mod:41]: do you need this line? `github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightskueuereceiver` doesnt exist upstream

Comment (rvasahu-amazon) [reply go.mod]: Versions updated.

I believe that line is necessary as the new module in our fork of the OTel collector repo is called `github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightskueuereceiver`.

File: go.mod [hunk 3]
```diff
@@ -7,40 +7,41 @@ replace github.com/influxdata/telegraf => github.com/aws/telegraf v0.10.2-0.2024
 // Replace with https://github.com/amazon-contributing/opentelemetry-collector-contrib, there are no requirements for all receivers/processors/exporters
 // to be all replaced since there are some changes that will always be from upstream
 replace (
-	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awscloudwatchlogsexporter => github.com/amazon-contributing/opentelemetry-collector-contrib/exporter/awscloudwatchlogsexporter v0.0.0-20241104203805-20919412150d
+	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awscloudwatchlogsexporter => github.com/amazon-contributing/opentelemetry-collector-contrib/exporter/awscloudwatchlogsexporter v0.0.0-20241114192544-cb251fe99cbb
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter => github.com/amazon-contributing/opentelemetry-collector-contrib/exporter/awsemfexporter v0.0.0-20241024185216-f0b80d5c67a5
```
Comment (sky333999) [inline go.mod:11]: Think we missed updating this for awsemfexporter

Comment (rvasahu-amazon) [reply go.mod:11]: Got it. Waiting for results on unit tests in current builds, will push fix after that.

**Review by chadpatel — APPROVED** [review]

**Review by sky333999 — APPROVED** [review]

**Pull request merged**